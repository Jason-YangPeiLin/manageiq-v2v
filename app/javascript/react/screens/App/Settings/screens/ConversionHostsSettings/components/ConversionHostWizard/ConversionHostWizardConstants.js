export const stepIDs = {
  locationStep: 'conversionHostWizardLocationStep',
  hostsStep: 'conversionHostWizardHostsStep',
  authenticationStep: 'conversionHostWizardAuthenticationStep',
  resultsStep: 'conversionHostWizardResultsStep'
};
