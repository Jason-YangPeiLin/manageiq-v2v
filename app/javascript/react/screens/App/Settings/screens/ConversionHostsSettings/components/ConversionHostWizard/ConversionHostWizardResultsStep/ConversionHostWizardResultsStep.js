import React from 'react';
import { Form } from 'patternfly-react';

const ConversionHostWizardResultsStep = () => (
  <Form className="form-horizontal">
    <h2>TODO: Results Step Contents</h2>
  </Form>
);

export default ConversionHostWizardResultsStep;
