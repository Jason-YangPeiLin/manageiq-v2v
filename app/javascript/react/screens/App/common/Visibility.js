import React from 'react';
import PropTypes from 'prop-types';
import { noop } from 'patternfly-react';

// This component has no role on the page, and will be hidden from screen readers (role="presentation").
// Renders the children wrapped in visibility: hidden if the condition is met.
// Either way, the children take up space, visibility will not change the flow of the layout.
// The onKeyUp handler and role are used to satisfy eslint-plugin-jsx-a11y.
const Visibility = ({ children, hidden }) => (
  <span style={{ visibility: hidden ? 'hidden' : '' }} onKeyUp={noop} role="presentation">
    {children}
  </span>
);

Visibility.propTypes = {
  children: PropTypes.node,
  hidden: PropTypes.bool
};

export default Visibility;
