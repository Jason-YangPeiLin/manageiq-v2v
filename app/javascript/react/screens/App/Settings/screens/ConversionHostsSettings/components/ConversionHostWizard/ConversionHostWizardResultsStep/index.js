import ConversionHostWizardResultsStep from './ConversionHostWizardResultsStep';

export default ConversionHostWizardResultsStep;
