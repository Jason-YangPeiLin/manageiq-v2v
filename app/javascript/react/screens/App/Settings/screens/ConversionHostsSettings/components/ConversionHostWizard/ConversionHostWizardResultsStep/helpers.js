import { OPENSTACK } from '../../../../../../../../../common/constants';
import { VDDK } from '../ConversionHostWizardConstants';

export const getConfigureConversionHostPostBodies = (locationStepValues, hostsStepValues, authStepValues) =>
  hostsStepValues.hosts.map(host => {
    const openstackSpecificProperties =
      locationStepValues.providerType === OPENSTACK ? { auth_user: authStepValues.openstackUser } : {};
    const vmwareAuthProperties =
      authStepValues.transformationMethod === VDDK
        ? { vmware_vddk_package_url: authStepValues.vddkLibraryPath }
        : { vmware_ssh_private_key: authStepValues.vmwareSshKey.body };
    return {
      name: host.name,
      resource_type: host.type,
      resource_id: host.id,
      ssh_key: authStepValues.conversionHostSshKey.body,
      ...openstackSpecificProperties,
      ...vmwareAuthProperties
    };
  });
