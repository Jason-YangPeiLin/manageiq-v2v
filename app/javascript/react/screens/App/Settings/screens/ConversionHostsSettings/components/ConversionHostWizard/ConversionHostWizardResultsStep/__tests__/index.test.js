import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { reducer as formReducer } from 'redux-form';

import { generateStore } from '../../../../../../../common/testReduxHelpers';
import settingsReducer, { initialState as settingsInitialState } from '../../../../../../SettingsReducer';
import ConversionHostWizardResultsStep from '../ConversionHostWizardResultsStep';
import ConversionHostWizardResultsStepContainer from '../index';
import { RHV } from '../../../../../../../../../../common/constants';
import { VDDK } from '../../ConversionHostWizardConstants';

describe('ConversionHostWizardLocationStep integration test', () => {
  const store = generateStore(
    { form: formReducer, settings: settingsReducer },
    {
      form: {
        conversionHostWizardLocationStep: { values: { providerType: RHV } },
        conversionHostWizardHostsStep: { values: { hosts: [{ id: '1', type: 'mock host type', name: 'Mock Host' }] } },
        conversionHostWizardAuthenticationStep: {
          values: {
            conversionHostSshKey: { body: 'mock conversion host SSH key body' },
            openstackCaCerts: { body: '' },
            transformationMethod: VDDK,
            vddkLibraryPath: 'mock VDDK path'
          }
        }
      },
      settings: settingsInitialState
    }
  );

  const mountComponent = () =>
    mount(
      <Provider store={store}>
        <ConversionHostWizardResultsStepContainer />
      </Provider>
    );

  it('should mount ConversionHostWizardLocationStep with mapStateToProps reduced', () => {
    const wrapper = mountComponent();

    // query the unconnected component to assert reduced props
    const component = wrapper.find(ConversionHostWizardResultsStep);
    expect(component.props()).toMatchSnapshot();
  });
});
