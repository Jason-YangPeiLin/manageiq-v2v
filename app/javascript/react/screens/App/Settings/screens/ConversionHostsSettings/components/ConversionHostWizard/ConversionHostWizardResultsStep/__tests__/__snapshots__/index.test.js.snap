// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConversionHostWizardLocationStep integration test should mount ConversionHostWizardLocationStep with mapStateToProps reduced 1`] = `
Object {
  "isPostingConversionHosts": true,
  "isRejectedPostingConversionHosts": false,
  "postBodies": Array [
    Object {
      "auth_user": "root",
      "conversion_host_ssh_private_key": "mock conversion host SSH key body",
      "name": "Mock Host",
      "resource_id": "1",
      "resource_type": "mock host type",
      "vmware_vddk_package_url": "mock VDDK path",
    },
  ],
  "postConversionHostsAction": [Function],
  "postConversionHostsResults": Array [],
  "postConversionHostsUrl": "/api/conversion_hosts",
}
`;
